### PR TITLE
#154117276Ft Create events blueprint

### PR DESCRIPTION
**Pull request by betsybeth**
##### Test using postman

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -58,94 +68,90 @@ def save_user(self):
...
-
-    def delete_rsvp(self):
-        """Deletes a rsvp of the event."""
-        db.session.delete(self)
-        db.session.commit()
-
-    def __repr__(self):
-        """Returns a representaion of the Rsvp class instance."""
-        return "<Rsvp: {}>".format(self.name)
+#
```
Comment (meshnesh) [inline app/models.py]: remove the commented code if not being used anymore

File: tests/test_events.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        second_result = self.client.post(
+            "/login",
+            data=json.dumps(self.user),
+            content_type='application/json')
+        return second_result
+
+    def test_create_event(self):
+        self.registration()
+        token_ = json.loads(self.login().data.decode())['token']
+        print('>>>>', token_)
```
Comment (meshnesh) [inline tests/test_events.py:48]: remove the print statement

File: tests/test_users.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        second_result = self.client.post("/login", data=json.dumps(self.user), content_type='application/json')
+        result = self.client.post("/logout", data=json.dumps(self.user), content_type='application/json')
+        self.assertIn('successfully logout' , str(result.data))
+        self.assertEqual(result.status_code,200)
+    def test_reset_password(self):
+        user_data = { "email":'test@email.com',
+                        'new_password':'098765432'}
+        result = self.client.post('/reset-password', data=json.dumps(user_data), content_type='application/json')
+        self.assertEqual(result.status_code, 201)
+        self.assertIn('Password reset successful', str(result.data))
```
Comment (meshnesh) [inline tests/test_users.py:53]: test whether when the user logs in with the old password they would get an error

File: tests/test_events.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        token_ = json.loads(self.login().data.decode())['token']
+        result = self.client.post('/create_event', data=json.dumps(self.event), content_type='application/json', headers={
+            'Authorization':token_
+        })
+        res = self.client.put('/events/<int:id>/', data=json.dumps(self.event), content_type='application/json',headers={
+            'Authorization':token_
+        })
+        self.assertEqual(res.status_code, 200)
+
+    def test_delete_event(self):
```
Comment (meshnesh) [inline tests/test_events.py:80]: Test if you fetch the event after deleting you won't find it

File: tests/test_config.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        self.assertTrue(app.config[SQLALCHEMY_DATABASE_URI] ==
+                        "postgresql://postgres:postgres@localhost/flask_api")
+
+
+class TestTestingConfig(TestCase):
+    def create_app(self):
+        app.config.from_object(app.config.TestingConfig)
+        return app
+
+    def test_app_is_testing(self):
```
Comment (meshnesh) [inline tests/test_config.py:21]: It's good practice to hide the url to the db, place it in the `.env` file then import it

File: app/models.py [hunk 6]
```diff
@@ -58,94 +68,90 @@ def save_user(self):
...
+# class Event(db.Model):
```
Comment (meshnesh) [inline app/models.py]: Remove the comments if the code is unused

File: app/__init__.py [hunk 7]
```diff
@@ -1,16 +1,19 @@
 from flask_api import FlaskAPI
-from flask_sqlalchemy import SQLAlchemy
+from .auth.views import auth_blueprint
+from .models import db
```
Comment (Rodgers-M) [inline app/__init__.py:3]: initialize the `db` here in the app then import it in the models.

File: app/events/views.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        name = json_dict.get('name')
+        description = json_dict.get('description')
+        category = json_dict.get('category')
+        date = json_dict.get('date')
+        location = json_dict.get('location')
+        if not event:
+            response = {
+                'message':'no bucketlist available'
+            }
+            return make_response(jsonify(response)), 400
```
Comment (Rodgers-M) [inline app/events/views.py:75]: i think you have a typo here, the `status code` should be `404`

File: app/models.py [hunk 9]
```diff
@@ -109,9 +114,16 @@ def delete_event(self):
...
+            'name': self.name,
+            'description': self.description,
+            'category': self.category,
+            'date': self.date,
+            'author': self.author,
+            'location': self.location
+        }
 
 
 class Rsvp(db.Model):
```
Comment (Rodgers-M) [inline app/models.py:129]: just a refactor here, how about moving the `name`, `email` and `phone_no` attributes to the `User` model then define a relationship to get the same?

**Review by Rodgers-M — COMMENTED** [review]
great work.

**Pull request merged**